apps/web/app/components/hooks/use-user.ts

```diff
@@ -0,0 +1,28 @@
+import { useAccount } from "wagmi";
+import useUserRoleCookie from "./use-user-role-cookie";
+import type { UserRole } from "@/common/user-role-schema";
+import type { Address } from "viem";
+
+function useUser({ required }: { required: false }): { address?: Address; role?: UserRole };
+function useUser(): { address: Address; role: UserRole };
+function useUser({
+  required,
+}: { required?: boolean } = {}):
+  | { address: Address; role: UserRole }
+  | { address?: Address; role?: UserRole } {
+  const { cookie: role } = useUserRoleCookie();
+  const { address } = useAccount();
+
+  if (required && (!address || !role)) {
+    throw new Error("User not logged in");
```

Comment:
Do we want to fail in here of we want to redirect to `/`?

Reply:
The middleware is in charge of redirecting to `/` if the user tries to access a "protected" page. This hook can be used safely in a protected page, knowing that the role and address should always be defined.

Reply:
I see. Then if we get to this point is actually an error. :+1: